From 8336a58086d063ad858eb9dd3e3c02122334f05e Mon Sep 17 00:00:00 2001
Message-Id: <8336a58086d063ad858eb9dd3e3c02122334f05e.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Damien Lespiau <damien.lespiau@intel.com>
Date: Thu, 3 Jul 2014 08:12:35 +0100
Subject: [PATCH 298/312] REVERTME [VPG]: drm/i915: Revert "drm/i915: Reject
 the pin ioctl on gen6+"

This reverts commit 02f6bcccf7c324115747aae2f0addd6af5d321cd.

The OA buffer can contain global data (in particular, not linked to a
context or a single batch execution) about GPU events (eg. hw context
switches, rc6 transitions, frequency changes, ...) and needs to be
mapped to GGTT. The pin ioctl provided a way to do that.

Admittedly, this change broke what seems to be a valid use case of
pinning a buffer in GGTT, even when PPGTT is used (which is the reason
invoked in the commit message).

Although this patch comes from the external mailing list it hasn't
been approved upstream, it is being pulled here to unblock the
performance team whilst a performance driver is being written.

Cc: Daniel Vetter <daniel.vetter@ffwll.ch>
Cc: Tomasz Madajczak <tomasz.madajczak@intel.com>
Cc: Adam Rutkowski <adam.j.rutkowski@intel.com>

Issue: GMINL-1888
Change-Id: I738ddd143341a206990653be1ceeb929a7514452
Signed-off-by: Damien Lespiau <damien.lespiau@intel.com>
Acked-by: Chris Wilson <chris@chris-wilson.co.uk> # who didn't like his tools
Signed-off-by: Rafael Barbalho <rafael.barbalho@intel.com>
Upstream-repo: http://patchwork.freedesktop.org/patch/28985/
Upstream-commit: <not applicable>
Upstream-as-of-tag: <not applicable>
---
 drivers/gpu/drm/i915/i915_gem.c |    3 ---
 1 file changed, 3 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_gem.c b/drivers/gpu/drm/i915/i915_gem.c
index b6e4d97..f6079ba 100644
--- a/drivers/gpu/drm/i915/i915_gem.c
+++ b/drivers/gpu/drm/i915/i915_gem.c
@@ -4175,9 +4175,6 @@ i915_gem_pin_ioctl(struct drm_device *dev, void *data,
 	struct drm_i915_gem_object *obj;
 	int ret;
 
-	if (INTEL_INFO(dev)->gen >= 6)
-		return -ENODEV;
-
 	ret = i915_mutex_lock_interruptible(dev);
 	if (ret)
 		return ret;
-- 
1.7.9.5

